deep_test.go

```diff
@@ -539,6 +539,77 @@ func TestSlice(t *testing.T) {
 	}
 }
 
+func TestSiblingSlices(t *testing.T) {
+	father := []int{1, 2, 3, 4}
+	a := father[0:3]
+	b := father[0:3]
+
+	diff := deep.Equal(a, b)
+	if len(diff) > 0 {
+		t.Error("should be equal:", diff)
+	}
+	diff = deep.Equal(b, a)
+	if len(diff) > 0 {
+		t.Error("should be equal:", diff)
+	}
+
+	a = father[0:3]
+	b = father[0:2]
+	diff = deep.Equal(a, b)
+	if diff == nil {
+		t.Fatal("no diff")
+	}
+	if len(diff) != 1 {
+		t.Error("too many diff:", diff)
+	}
+	if diff[0] != "slice[2]: 3 != <no value>" {
+		t.Error("wrong diff:", diff[0])
+	}
+
+	a = father[0:2]
+	b = father[0:3]
+
+	diff = deep.Equal(a, b)
+	if diff == nil {
+		t.Fatal("no diff")
+	}
+	if len(diff) != 1 {
+		t.Error("too many diff:", diff)
+	}
+	if diff[0] != "slice[2]: <no value> != 3" {
+		t.Error("wrong diff:", diff[0])
+	}
+
+	a = father[0:2]
+	b = father[2:4]
+
+	diff = deep.Equal(a, b)
+	if diff == nil {
+		t.Fatal("no diff")
+	}
+	if len(diff) != 2 {
+		t.Error("too many diff:", diff)
+	}
+	if diff[0] != "slice[0]: 1 != 3" {
+		t.Error("wrong diff:", diff[0])
+	}
+	if diff[1] != "slice[1]: 2 != 4" {
+		t.Error("wrong diff:", diff[1])
+	}
+
+	a = father[0:0]
+	b = father[1:1]
+
+	diff = deep.Equal(a, b)
+	if len(diff) > 0 {
```

Comment:
@risteli Shouldn't `a` and `b` be different here? Unless I'm mistaken, `a` will equal 1 and `b` equal `2`?

Reply:
I'm not sure what I was thinking back on Jan 23, 2018, but today I see they'd both be len=0.